cmd/fetchd/cmd/genreplace.go

```diff
@@ -0,0 +1,164 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+)
+
+const (
+	flagNewDescription = "new-description"
+)
+
+// ReplaceGenesisValuesCmd returns replace-genesis-values cobra Command.
+func ReplaceGenesisValuesCmd(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "replace-genesis-values [new-denom] [old-denom]",
+		Short: "Replace the major values within genesis.json, such as native denom",
+		Long: `Replace the major values within genesis.json, such as native denom.
+`,
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			cdc := clientCtx.Codec
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
+
+			newDesc, err := cmd.Flags().GetString(flagNewDescription)
+			if err != nil {
+				return err
+			}
+
+			inputData := replacementData{
+				NewBaseDenom:   args[0],
+				OldBaseDenom:   args[1],
+				NewDenom:       fmt.Sprintf("a%s", args[0]),
+				OldDenom:       fmt.Sprintf("a%s", args[1]),
+				NewDescription: newDesc,
+				OldAddrPrefix:  "fetch",
+				NewAddrPrefix:  "asi",
+			}
+
+			genFile := config.GenesisFile()
+
+			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+
+			//authGenState := authtypes.GetGenesisStateFromAppState(cdc, appState) TODO: amend account prefix here, potentially
+
+			bankGenState := banktypes.GetGenesisStateFromAppState(cdc, appState)
+			replaceDenomMetadata(&inputData, bankGenState)
+
+			bankGenStateBz, err := cdc.MarshalJSON(bankGenState)
+			if err != nil {
+				return fmt.Errorf("failed to marshal auth genesis state: %w", err)
+			}
+
+			appState[banktypes.ModuleName] = bankGenStateBz
+
+			appStateJSON, err := json.Marshal(appState)
+			if err != nil {
+				return fmt.Errorf("failed to marshal application genesis state: %w", err)
+			}
+
+			appStateStr := string(appStateJSON)
+			for _, target := range []string{"denom", "bond_denom", "mint_denom", "base_denom", "base"} {
+				re := regexp.MustCompile(fmt.Sprintf(`("%s"\s*:\s*)"%s"`, target, inputData.OldDenom))
```

Comment:
Just note: Precompilation of regexes makes sense if regexp is executed repeatidly many times, and it is done just once per process lifetime, usually creating & precompiling regex statements in global namespace/scope as global contants/variables.

But in this case it does NOT make sense to do anything in global namespace/scope, since the each regexp is used just **once** for this single purpose and for nothing else.